### PR TITLE
fix: change metadata of models and fix minor things

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

- add `input` to the `asyncapi` metadata of models.
- `source` in model metadata now points to the source of file/document, not to the input.
- to boolean values we cannot assign things like `x-parser-schema-id` so I will add proper if in `setExtension` function.
- catch global errors from `validate` and `parse` functions.
- remove `version()` method from `Extension` model - we don't have clarified it in spec, so we shouldn't add that in current api.
- revert some changes in the `additionalItems` and `additionalProperties` methods from old Schema model - we havw to consistent with old logic.
- add unit tests.

cc @smoya

### CODE REVIEW

File: src/old-api/schema.ts [hunk 1]
```diff
@@ -116,20 +116,15 @@ export class Schema extends SpecificationExtensionsModel<v2.AsyncAPISchemaObject
...
     return this.__createChild(additionalProperties);
   }
 
   additionalItems() {
-    if (typeof this._json === 'boolean') return this._json;
     const additionalItems = this.__get('additionalItems');
-    if (typeof additionalItems === 'boolean') return additionalItems;
-    if (additionalItems === undefined) return true;
-    if (additionalItems === null) return false;
+    if (additionalItems === undefined || additionalItems === null) return;
```
Comment (magicmatatjahu) [inline src/old-api/schema.ts:127]: We have to be consistent with implementation in v1 parser - current logic in `next-major` has breaking change in `additionalItems` and `additionalProperties` methods and some templates won't work.

Comment (magicmatatjahu) [reply src/old-api/schema.ts:127]: https://github.com/asyncapi/parser-js/blob/master/lib/models/schema.js#L221
https://github.com/asyncapi/parser-js/blob/master/lib/models/schema.js#L231

File: src/models/extension.ts [hunk 2]
```diff
@@ -2,6 +2,5 @@ import type { BaseModel } from './base';
 
 export interface ExtensionInterface<T = any> extends BaseModel {
   id(): string;
-  version(): string;
```
Comment (smoya) [inline src/models/extension.ts:5]: Please ensure this change is proposed in https://github.com/asyncapi/parser-api/pull/71

Comment (magicmatatjahu) [reply src/models/extension.ts:5]: Done https://github.com/asyncapi/parser-api/pull/71#discussion_r1006894610 :)

File: src/utils.ts [hunk 3]
```diff
@@ -86,6 +90,22 @@ export function toJSONPathArray(jsonPath: string): Array<string | number> {
...
+        message: `${message}. Name: ${err.name}, message: ${err.message}, stack: ${err.stack}`,
+        path: [],
+        severity: DiagnosticSeverity.Error,
+        range,
+      }
+    ];
+  }
+  return [];
+}
+
```
Comment (smoya) [inline src/utils.ts:108]: Super minor but readability improvement in my opinion:

```suggestion
export function createUncaghtDiagnostic(err: unknown, message: string, document?: Document): Diagnostic[] {
  if (!err instanceof Error) {
    return [];
  }
  
  const range: Diagnostic['range'] = document ? document.getRangeForJsonPath([]) as Diagnostic['range'] : Document.DEFAULT_RANGE;
  return [
    {
      code: 'uncaught-error',
      message: `${message}. Name: ${err.name}, message: ${err.message}, stack: ${err.stack}`,
      path: [],
      severity: DiagnosticSeverity.Error,
      range,
    }
  ];  
}
```

**Review by smoya — APPROVED** [review]
LGTM!! Just one very minor, feel free to discard

**Comment by magicmatatjahu** [comment]
@smoya Thanks!

**Pull request merged**